tests/src/JIT/Methodical/tailcall_v4/hijacking.il

```diff
@@ -263,22 +264,6 @@
                                                                                      native int)
     IL_0032:  newobj     instance void [System.Threading.Thread]System.Threading.Thread::.ctor(class [System.Threading.Thread]System.Threading.ThreadStart)
     IL_0037:  stelem.ref
-    IL_0038:  ldloc.s    V_5
-    IL_003a:  ldc.i4.2
-    IL_003b:  ldloc.0
-    IL_003c:  ldftn      instance void Repro::TailCallThread()
-    IL_0042:  newobj     instance void [System.Threading.Thread]System.Threading.ThreadStart::.ctor(object,
-                                                                                     native int)
-    IL_0047:  newobj     instance void [System.Threading.Thread]System.Threading.Thread::.ctor(class [System.Threading.Thread]System.Threading.ThreadStart)
-    IL_004c:  stelem.ref
-    IL_004d:  ldloc.s    V_5
-    IL_004f:  ldc.i4.3
-    IL_0050:  ldloc.0
-    IL_0051:  ldftn      instance void Repro::TailCallThread()
-    IL_0057:  newobj     instance void [System.Threading.Thread]System.Threading.ThreadStart::.ctor(object,
-                                                                                     native int)
-    IL_005c:  newobj     instance void [System.Threading.Thread]System.Threading.Thread::.ctor(class [System.Threading.Thread]System.Threading.ThreadStart)
-    IL_0061:  stelem.ref
```

Comment:
I have modernized this test:
1. I have added another field to the `ForceHelper` struct so it also forces helper on SysV ABI.
2. This test worked by running 1 thread continuously GC'ing and 3 threads in two functions that tailcalled each other mutually. It is trying to test that the tailcalling logic correctly works in the face of return address hijacking. I had this test time out so I have changed it to 1 GC'ing thread and 1 tailcalling thread so that we can hopefully run it without timeout. I have verified manually that it still fails almost instantaneously when the return address hijacking logic is not handled correctly.

Reply:
3. I have also changed it to run for 30 iterations of 1 second sleeps instead of 60 iterations of 1 second sleeps since I saw a timeout of this test in CI, even with the other changes.